ps_facetedsearch.php

```diff
@@ -1354,511 +1335,16 @@ private function checkLinksRewrite($params)
-     * Attribute group form
-     *
-     * @param array $params
-     *
-     * @return string
-     */
-    public function hookAttributeGroupForm(array $params)
-    {
-        $values = [];
-        $isIndexable = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
-            'SELECT `indexable`
-            FROM ' . _DB_PREFIX_ . 'layered_indexable_attribute_group
-            WHERE `id_attribute_group` = ' . (int) $params['id_attribute_group']
-        );
-
-        if ($isIndexable === false) {
-            $isIndexable = true;
-        }
-
-        if ($result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
-            'SELECT `url_name`, `meta_title`, `id_lang` FROM ' . _DB_PREFIX_ . 'layered_indexable_attribute_group_lang_value
-            WHERE `id_attribute_group` = ' . (int) $params['id_attribute_group']
-        )) {
-            foreach ($result as $data) {
-                $values[$data['id_lang']] = ['url_name' => $data['url_name'], 'meta_title' => $data['meta_title']];
-            }
-        }
-
-        $this->context->smarty->assign([
-            'languages' => Language::getLanguages(false),
-            'default_form_language' => (int) $this->context->controller->default_form_language,
-            'values' => $values,
-            'is_indexable' => (bool) $isIndexable,
-        ]);
-
-        return $this->display(__FILE__, 'attribute_group_form.tpl');
-    }
-
-    /**
-     * After save attribute
-     *
-     * @param array $params
-     */
-    public function hookAfterSaveAttribute(array $params)
-    {
-        if (empty($params['id_attribute'])) {
-            return;
-        }
-
-        Db::getInstance()->execute(
-            'DELETE FROM ' . _DB_PREFIX_ . 'layered_indexable_attribute_lang_value
-            WHERE `id_attribute` = ' . (int) $params['id_attribute']
-        );
-
-        foreach (Language::getLanguages(false) as $language) {
-            $seoUrl = Tools::getValue('url_name_' . (int) $language['id_lang']);
-
-            if (empty($seoUrl)) {
-                $seoUrl = Tools::getValue('name_' . (int) $language['id_lang']);
-            }
-
-            Db::getInstance()->execute(
-                'INSERT INTO ' . _DB_PREFIX_ . 'layered_indexable_attribute_lang_value
-                (`id_attribute`, `id_lang`, `url_name`, `meta_title`)
-                VALUES (
-                ' . (int) $params['id_attribute'] . ', ' . (int) $language['id_lang'] . ',
-                \'' . pSQL(Tools::link_rewrite($seoUrl)) . '\',
-                \'' . pSQL(Tools::getValue('meta_title_' . (int) $language['id_lang']), true) . '\')'
-            );
-        }
-        $this->invalidateLayeredFilterBlockCache();
-    }
-
-    /**
-     * After delete attribute
+     * Dispatch hooks
      *
-     * @param array $params
+     * @param string $methodName
+     * @param array $arguments
      */
-    public function hookAfterDeleteAttribute(array $params)
+    public function __call($methodName, array $arguments)
```

Comment:
@PierreRambaud using __call in modules was dangerous for many reasons in 1.6, for example you couldn't export theme with this kind of module using "Export theme" feature and sometimes there was an issue while sorting module in Modules -> Positions, i suggest to check this out :)

Reply:
Can you tell me more? Should be a bug because having a file with 3k lines because of this is a really bad idea.

Reply:
And we should be able to use everything we want in module 🤔  (at least having the architecture we want)

Reply:
I agree we should, but there is so much magic in PrestaShop ^^ It reminds me of the translation system.. which scans the modules' code to extract the call to `trans` or `l` methods to retrieve the translation keys.. So if you use a variable it doesn't work.

I don't know the module system or legacy feature well enough to understand how it could be a problem. It's worth checking if everything works fine as @kpodemski suggested.

By the way which version does this new version of the module target? Only 1.7.x+ or 1.6 as well?